server/blob/blob-joining/src/test/java/org/apache/james/blob/joining/JoiningBlobStoreTest.java

```diff
@@ -0,0 +1,213 @@
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.joining;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.BlobStoreContract;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.util.CompletableFutureUtil;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+class JoiningBlobStoreTest implements BlobStoreContract {
+
+    private static class ThrowingBlobStore implements BlobStore {
+
+        @Override
+        public CompletableFuture<BlobId> save(byte[] data) {
+            if (Arrays.equals(data, BLOB_CONTENT_CAUSES_THROWING_DIRECTLY)) {
+                throw new RuntimeException("not supported");
+            }
+
+            return CompletableFutureUtil.exceptionallyFuture(new RuntimeException("not supported"));
+        }
+
+        @Override
+        public CompletableFuture<BlobId> save(InputStream data) {
+            try {
+                if (Arrays.equals(IOUtils.toByteArray(data), BLOB_CONTENT_CAUSES_THROWING_DIRECTLY)) {
+                    throw new RuntimeException("not supported");
+                }
+
+                return CompletableFutureUtil.exceptionallyFuture(new RuntimeException("not supported"));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public CompletableFuture<byte[]> readBytes(BlobId blobId) {
+            return CompletableFutureUtil.exceptionallyFuture(new RuntimeException("not supported"));
+        }
+
+        @Override
+        public InputStream read(BlobId blobId) {
+            throw new RuntimeException("not supported");
+        }
+    }
+
+    private static final HashBlobId.Factory BLOB_ID_FACTORY = new HashBlobId.Factory();
+    private static final MemoryBlobStore PRIMARY_BLOB_STORE = new MemoryBlobStore(BLOB_ID_FACTORY);
+    private static final MemoryBlobStore SECONDARY_BLOB_STORE = new MemoryBlobStore(BLOB_ID_FACTORY);
```

Comment:
These  blob store should not be static but should be class fields.

That way you should not need to clear them.